python/lsst/ts/observatory/control/generic_camera.py

```diff
@@ -0,0 +1,157 @@
+# This file is part of ts_observatory_control.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["GenericCamera"]
+
+import typing
+import logging
+
+from lsst.ts import salobj
+
+from .base_camera import BaseCamera
+
+
+class GenericCamera(BaseCamera):
+    """Generic camera high level class.
+
+    This class is designed to operate the generic camera CSC using the same
+    `BaseCamera` interface.
+
+    Parameters
+    ----------
+    index : `int`
+        Generic camera index.
+    domain : `salobj.Domain`
+        Domain to use of the Remotes. If `None`, create a new domain.
+    log : `logging.Logger`
+        Optional logging class to be used for logging operations. If `None`,
+        creates a new logger. Useful to use in salobj.BaseScript and allow
+        logging in the class use the script logging.
+    intended_usage: `int`
+        Optional integer that maps to a list of intended operations. This is
+        used to limit the resources allocated by the class by gathering some
+        knowledge about the usage intention. By default allocates all
+        resources.
+    instrument_setup_attributes : `list` [`str`], optional
+        Optional names of the attributes needed to setup the instrument for
+        taking an exposure. This is used to check for bad input with functions
+        calls.
+    tcs_ready_to_take_data : `coroutine`
+        A coroutine that waits for the telescope control system to be ready
+        to take data.
+    """
+
+    def __init__(
+        self,
+        index: int,
+        domain: typing.Optional[salobj.Domain] = None,
+        log: typing.Optional[logging.Logger] = None,
+        intended_usage: typing.Optional[int] = None,
+        instrument_setup_attributes: typing.Optional[typing.List[str]] = None,
```

Comment:
With Python 3.10 you can use `list` instead of `typing.List`. Please change this and all other occurances.